packages/app-content-pages/server/configure-next-server.spec.js

```diff
@@ -0,0 +1,76 @@
+const proxyquire = require('proxyquire')
+const sinon = require('sinon')
+
+const configureNextServer = require('./configure-next-server')
+
+const mockApp = {
+  getRequestHandler: () => sinon.spy(),
+  setAssetPrefix: sinon.spy()
+}
+
+const setCacheHeaders = sinon.spy()
+
+describe('configureNextServer', function () {
+  it('should exist', function () {
+    expect(configureNextServer).to.be.a('function')
+  })
+
+  describe('assetPrefix setting', function () {
+    it('should default to an empty string', function () {
+      const assetPrefix = ''
+      configureNextServer(mockApp)
+      const result = mockApp.setAssetPrefix.calledWith(assetPrefix)
+      expect(result).to.be.true()
```

Comment:
We've been using sinon-chai in other tests. It's a bit more readable, so this expectation instead could be written as: `expect(mockApp.setAssetPrefix).to.be.calledWith(assetPrefix)`

Reply:
```js
const prefixSpy = mockApp.setAssetPrefix.withArgs(assetPrefix)
expect(prefixSpy).to.have.been.called()
```
 would work too. I've been using `spy.withArgs()` in recent tests but I can't remember why it's preferred over `.calledWith()`. Maybe because it gives you a spy with call history etc. for those specific arguments?

Reply:
Yeah, I think it matches the arguments first and then returns the call history only for that. I think it's useful if you already know that there may be many calls the spy is counting but only X for a specific argument and that's what you want to test for. However, I think in this case, they're basically doing the same thing.